src/wallet/wallet.cpp

```diff
@@ -1689,6 +1689,11 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
         if (max_height && block_height >= *max_height) {
             break;
         }
+        if (block_height > GetLastBlockHeight()) {
```

Comment:
ASAN CI: "wallet/wallet.cpp:1692:28: error: calling function 'GetLastBlockHeight' requires holding mutex 'cs_wallet' exclusively [-Werror,-Wthread-safety-analysis]"

Reply:
added the lock for this, waiting on CI